datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -873,6 +873,36 @@ macro_rules! get_data_page_statistics {
                     Decimal128Array::from_iter([<$stat_type_prefix Decimal128DataPageStatsIterator>]::new($iterator).flatten()).with_precision_and_scale(*precision, *scale)?)),
                 Some(DataType::Decimal256(precision, scale)) => Ok(Arc::new(
                     Decimal256Array::from_iter([<$stat_type_prefix Decimal256DataPageStatsIterator>]::new($iterator).flatten()).with_precision_and_scale(*precision, *scale)?)),
+                Some(DataType::Time32(unit)) => {
+                    Ok(match unit {
+                        TimeUnit::Second =>  Arc::new(Time32SecondArray::from_iter(
+                            [<$stat_type_prefix Int32DataPageStatsIterator>]::new($iterator).flatten(),
+                        )),
+                        TimeUnit::Millisecond => Arc::new(Time32MillisecondArray::from_iter(
+                            [<$stat_type_prefix Int32DataPageStatsIterator>]::new($iterator).flatten(),
+                        )),
+                        _ => {
+                            let len = $iterator.count();
```

Comment:
I have referenced `get_statistics` macro to decide on returing a null array, my other idea was to conclued with `unimplemented!()`.

Reply:
You beat me to it ;-). 

I think this can just be:
```Rust
                        _ => new_empty_array(&DataType::Time32(unit.clone())),
```

And then you need to add new_empty_array into the use statement. Time64 gets a similar treatment.

Reply:
Much cleaner. Thanks for the idea @efredine